Add Couchbase vendor profile

1tylermitchell opened the pull request:
New vendor/product profile information. And a small typo fix for the main listing page.

1tylermitchell commented:
I'm reviewing the validation issues.

svonava commented:
> I'm reviewing the validation issues.

Thank you! Besides that, if you could provide more source_url references for the partial/full support claims, that would be also very helpful for the reviewers and community broadly - folks love to dive into docs/examples when exploring how you support each given feature.

svonava commented:
> Thanks @svonava - also, could Erlang be added to the dev langs schema for validate options?

Added, make sure to lowercase all the options / see how the other files do this.

1tylermitchell commented:
thanks so much

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  "oss": {
+    "support": "partial",
+    "source_url": "https://www.couchbase.com/blog/couchbase-adopts-bsl-license/",
+    "comment": ""
```
> Vector search currently only part of Enterprise and DBaaS editions

Could you add this comment here?

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  },
+  "dev_languages": {
+    "value": [
+      "Python", "c++", "erlang", "go", "java"
```
please limit this list to the 1 or 2 main languages that the DB is written in. Like Daniel mentioned, they need to be in lowercase as specified in the enum here: https://github.com/superlinked/VectorHub/blob/49931762584d2a4f91cbc063d348dc390563a310/docs/tools/vdb_table/vendor.schema.json#L86-L97

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+    "value": [
+      "Python", "c++", "erlang", "go", "java"
+    ],
+    "source_url": "https://docs.couchbase.com/home/sdk.html",
+    "comment": "SDK client support: Python, C++, C, Go, Java, Kotlin, Node, PHP, Ruby, Scala, .Net"
```
please remove these. This column is not about SDK languages

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  "vector_launch_year": 2024,
+  "metadata_filter": {
+    "support": "full",
+    "source_url": "https://www.prnewswire.com/news-releases/couchbase-announces-new-features-to-accelerate-ai-powered-adaptive-applications-for-customers-302074941.html",
```
this needs to be a link to the documentation where it is shown how to filter on various attributes (<,>,== etc.)

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  },
+  "geo_search": {
+    "support": "full",
+    "source_url": "https://docs.couchbase.com/cloud/search/geo-search-ui.html",
```
I see that you are able to create the index and query via UI. Is it also possible to do the same through the SDK?

1tylermitchell replied:
Yes it is possible via SDK too.

dhruv-anand-aintech replied:
would be good to provide a link to the exact function in the documentation for that.

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+    "comment": ""
+  },
+  "embeddings_text": {
+    "support": "partial",
+    "source_url": "https://docs.couchbase.com/cloud/vector-search/run-vector-search-ui.html#large",
+    "comment": "via external embedding models"
+  },
+  "embeddings_image": {
+    "support": "partial",
+    "source_url": "",
+    "comment": "via external embedding models"
+  },
+  "embeddings_structured": {
+    "support": "partial",
+    "source_url": "",
+    "comment": "via external embedding models"
```
please mark as none since these columns are about operations that happen in the DB

1tylermitchell replied:
done, sorry missed these last few comments as they were hidden for some reason

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+    "comment": "via external embedding models"
+  },
+  "rag": {
+    "support": "partial",
+    "source_url": "https://docs.couchbase.com/cloud/vector-search/run-vector-search-ui.html#large",
+    "comment": "via external LLMs"
```
this refers to being able to generate the answers from the DB (via deep integration of LLM APIs. Have a look at the other DBs which have this marked as supported

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  },
+  "multi_tenancy": {
+    "support": "full",
+    "source_url": "",
```
adding a source here would be good

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  },
+  "disk_index": {
+    "support": "full",
+    "source_url": "",
```
adding a source here would be good

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+    "source_url": "",
+    "comment": ""
+  },
+  "ephemeral": {
+    "support": "full",
+    "source_url": "https://docs.couchbase.com/server/current/learn/buckets-memory-and-storage/buckets.html#replication-dcp-and-xdcr",
+    "comment": ""
```
this column refers to the ability to have an index without a server (on client side itself)

dhruv-anand-aintech replied:
removing

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+  },
+  "int8_quantization": {
+    "support": "partial",
+    "source_url": "",
```
adding a source here would be good

dhruv-anand-aintech commented on docs/tools/vdb_table/data/couchbase.json:
```diff
@@ -0,0 +1,181 @@
+    "source_url": "",
+    "comment": "Automatic algorithm selection between: IdMap2,Flat, IVF,Flat, IVF,SQ8"
+  },
+  "github_stars": {
+    "value": 6000,
+    "source_url": "https://github.com/couchbase",
+    "comment": "",
```
this field is only for repositories which are open-source and contain the vector search functionality within them

dhruv-anand-aintech reviewed (changes requested):
Thanks for adding the DB to this list!

I have a few comments that need to be addressed before it can be merged.

1tylermitchell commented:
Thanks for the feedback! Addressed most issues, awaiting on a few docs links for a couple newer features, but otherwise ready. Will keep up to date when I get more links to add.

dhruv-anand-aintech commented:
merging now

1tylermitchell commented:
thank you!
